components/territory-form.js

```diff
@@ -28,12 +29,53 @@ export default function TerritoryForm ({ sub }) {
         }
       }`
   )
+  const [unarchiveTerritory] = useMutation(
+    gql`
+      mutation unarchiveTerritory($name: String!, $desc: String, $baseCost: Int!,
+        $postTypes: [String!]!, $allowFreebies: Boolean!, $billingType: String!,
+        $billingAutoRenew: Boolean!, $moderated: Boolean!, $hash: String, $hmac: String, $nsfw: Boolean!) {
+          unarchiveTerritory(name: $name, desc: $desc, baseCost: $baseCost,
+            postTypes: $postTypes, allowFreebies: $allowFreebies, billingType: $billingType,
+            billingAutoRenew: $billingAutoRenew, moderated: $moderated, hash: $hash, hmac: $hmac, nsfw: $nsfw) {
+          name
+        }
+      }`
+  )
+
+  const [archived, setArchived] = useState(false)
+
+  const validate = async (values) => {
+    // validate schema manually so we can handle the error if sub is archived.
+    // in that case, we want to show a warning text but not prevent form submission.
+
+    const errors = {}
+    const schema = territorySchema({ client, me })
+
+    try {
+      await schema.validate(values, { abortEarly: false })
+    } catch (e) {
+      if (!(e instanceof ValidationError)) throw e
+      for (const { path, errors: [message] } of e.inner) {
+        // use the first error for each field so 'archived' has higher precedence over 'taken' for sub name
+        errors[path] ??= message
+      }
+    }
+
+    if (errors.name === 'archived') {
+      delete errors.name
+      setArchived(true)
+    } else {
+      setArchived(false)
+    }
+
+    return errors
+  }
```

Comment:
I wondered if I should refactor this since it's kind of weird to run validation just to cast an error into a visual warning which does not prevent submission.

But since this didn't require a lot of changes and it's pretty isolated, I think it's fine.

An alternative I considered was to use `useEffect` (obviously :smile:) but then I would have to do something weird to get access to the name value which isn't readily available outside the `Form(ik)` component.

So I think hooking into validation was the right choice.

Reply:
It's using composition so it's at least easy to understand.

IIRC formik does give us access to the error messages on fields, so my impulse would be to use that, but I assume you already tried.

Reply:
> IIRC formik does give us access to the error messages on fields, so my impulse would be to use that, but I assume you already tried.

No, I didn't :eyes: Do you mean [field-level validation](https://formik.org/docs/guides/validation#field-level-validation) or [`ErrorMessage`](https://formik.org/docs/api/errormessage)?

Reply:
I mean this: https://formik.org/docs/api/useField#fieldmetapropsvalue

Reply:
Mhh, but I am doing what I am doing here to not prevent form submission if a territory is archived, only show the warning. So it can't be an error that Formik concerns that would get returned by [`useField`](https://formik.org/docs/api/useField#fieldmetapropsvalue).

So afaict, either a) I use validation to check for archived territories but then need to prevent submission getting blocked or b) don't use validation but check for archived territories in some other way (probably `useEffect`)

Reply:
Oh true. Another alternative is setting up an `onChange` handler that sets this state ... which might be better than overloading validation like this?